index.js

```diff
@@ -52,12 +52,26 @@ module.exports = function (options) {
       process.exit(1)
     }, 10000)
 
-    cluster.disconnect(function () {
-      restarting = false
-      clearTimeout(timeout)
-      logger.info('Cluster shutdown cleanly')
-      process.exit(0)
-    })
+    var workersAllExited= function() {
+      var runningWorker = _.find(workers, function (worker) {
```

Comment:
I think this could be shortened to `return _.every(workers, {state: 'exited'})`

https://github.com/lodash/lodash/blob/3.10.1/doc/README.md#_everycollection-predicate_identity-thisarg